internal/node/node_repositories.bzl

```diff
@@ -436,12 +443,25 @@ if %errorlevel% neq 0 exit /b %errorlevel%
             "TEMPLATED_is_windows": "true" if is_windows else "false",
             "TEMPLATED_node_actual": node_entry,
             "TEMPLATED_node_bin_actual": node_exec_label,
-            "TEMPLATED_node_dir": NODE_DIR,
             "TEMPLATED_npm_actual": npm_node_repositories_entry,
+            "TEMPLATED_vendored_node": "true" if repository_ctx.attr.vendored_node else "false",
             "TEMPLATED_yarn_actual": yarn_node_repositories_entry,
         },
     )
-    result = repository_ctx.execute([node_entry, "generate_build_file.js"])
+    host_os = os_name(repository_ctx)
+    if host_os in repository_ctx.attr.name or repository_ctx.attr.name == "nodejs":
+        # We have to use the relative path here otherwise bazel reports a cycle
+        result = repository_ctx.execute([node_entry, "generate_build_file.js"])
+    else:
+        # NOTE: Ideally we would not need this logic here and could just depend on the @nodejs//:node_bin alias but it is not possible.
+        # See: https://github.com/bazelbuild/bazel/issues/8674
+        node_path = "node.exe" if is_windows_os else "bin/node"
```

Comment:
Took me a second to see what was going on here. This is a check if the nodejs repo being setup is the same as the host since we have to run `node generate_build_files.js` so if the nodejs repo is not the same as the host, the @nodejs host repo node executable is used. Can add a bit more explanation?

Another PR we should remove generate_build_file.js altogether if possible and just write a BUILD file using `repository_ctx.template` directly instead of generating ` generate_build_file.js`. Not sure why that was put in in the first place but I don't see anything that is does that starlark couldn't handle.

Reply:
Yep agreed would be much nicer if we could get rid of the `generate_build_file.js`. Extended the comment.